File: Zamba/ZambaWeb2013/zamba2010/Zamba.WebInside/MasterPage.master.vb
'Imports Zamba.Core
'Imports Zamba.Services
Imports System.Collections.Generic

Partial Class MasterPage

    Inherits System.Web.UI.MasterPage

    Protected Sub Page_Init(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Init

        If Session("UserId") Is Nothing Then
            'FormsAuthentication.RedirectToLoginPage()
            'Response.End()
        End If

        'icono del titulo
        lnkWebIcon.Attributes.Add("href", "~/App_Themes/" & Page.Theme & "/Images/WebIcon.ico")
    End Sub

    Protected Sub Page_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load
        'Try
        '    If Not Page.IsPostBack Then
        '        If Not Session("User") Is Nothing Then
        '            Dim user As IUser
        '            user = Session("User")
        '            Me.lblUsuarioActual2.Text = user.Name
        '            Me.lblUsuarioActual.Text = user.Name
        '            hdnUserID.Value = user.ID.ToString()
        '            Dim zopt As New ZOptBusiness
        '            Dim link As String = zopt.GetValue("WebHomeLink")
        '            Dim target As String = zopt.GetValue("WebHomeTarget")
        '            zopt = Nothing
        '            Dim sUserPref As New SUserPreferences
        '            hdnRefreshWF.Value = sUserPref.getValue("WebRefreshWFTab", Sections.WorkFlow, False)

        '            If String.IsNullOrEmpty(link) Then
        '                hdnLink.Value = HttpContext.Current.Request.Url.ToString()
        '            Else
        '                hdnLink.Value = link
        '            End If

        '            If String.IsNullOrEmpty(target) Then
        '                hdnTarget.Value = "_self"
        '            Else
        '                hdnTarget.Value = target
        '            End If
        '            'Me.UC_WFExecution.TaskID = Task_ID

        '            Session("ListOfTask") = Nothing

        '            'Actualiza el timemout
        '            Dim rights As SRights = New SRights()
        '            Dim type As Int32 = 0
        '            If user.WFLic Then type = 1
        '            If (user.ConnectionId > 0) Then
        '                Dim SUserPreferences As New SUserPreferences()
        '                rights.UpdateOrInsertActionTime(user.ID, user.Name, user.puesto, user.ConnectionId, Int32.Parse(SUserPreferences.getValue("TimeOut", Sections.UserPreferences, "30")), type)
        '                SUserPreferences = Nothing
        '            Else
        '                Response.Redirect("~/Views/Security/LogIn.aspx")
        '            End If
        '            rights = Nothing
        '        End If
        '    End If

        '    hdnUserID.Value = Session("UserId")
        '    hdnConnectionId.Value = Session("ConnectionId")
        '    hdnComputer.Value = Session("ComputerNameOrIP")
        'Catch ex As Exception
        '    ZClass.raiseerror(ex)
        'End Try
    End Sub

    Protected Sub MasterLogout(ByVal sender As Object, ByVal e As System.EventArgs)

    End Sub

    Public Shared Function GetAppRootUrl(ByVal endSlash As Boolean) As String
        Dim host As String = HttpContext.Current.Request.Url.GetLeftPart(UriPartial.Authority)
        Dim appRootUrl As String = HttpContext.Current.Request.ApplicationPath
        If Not appRootUrl.EndsWith("/") Then
            'a virtual
            appRootUrl += "/"
        End If
        If Not endSlash Then
            appRootUrl = appRootUrl.Substring(0, appRootUrl.Length - 1)
        End If
        Return host + appRootUrl
    End Function
End Class

File: Zamba/Zamba.Web/Zamba.Web/Zamba.Web/MasterPage.master.cs

using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Web;
using System.Web.Mvc;
using System.Web.Script.Serialization;
using System.Web.Security;
using System.Web.UI;
using System.Web.UI.WebControls;
using Zamba.Core;
using Zamba.Membership;
using Zamba.Services;

namespace Zamba.Web
{
    partial class MasterPage : System.Web.UI.MasterPage
    {
        public bool loadChatZmb { get; set; }
        public bool loadGlobalSearch { get; set; }

        public string WebServiceResultsURL;
       

        protected void Page_Init(object sender, System.EventArgs e)
        {
            WebServiceResultsURL = ZOptBusiness.GetValueOrDefault("WSResultsUrl", "http://www.zamba.com.ar/zambastardocWS");
            //icono del titulo
            lnkWebIcon.Attributes.Add("href", "~/App_Themes/" + Page.Theme + "/Images/WebIcon.jpg");

            //if (Membership.MembershipHelper.CurrentUser != null)
            //{
            //    SWorkflow Sworkflow = new SWorkflow();

            //    var WorkFlows = Sworkflow.GetUserWFIdsAndNamesWithSteps(Membership.MembershipHelper.CurrentUser.ID);

            //    if (WorkFlows != null && WorkFlows.Count > 0)
            //    {
            //        ScriptManager.RegisterClientScriptBlock(this, GetType(), "TaskCount", "$(document).ready(function () { LoadStepsCounts(); });", true);
            //    }
            //}
        }
        protected void Page_Load(object sender, System.EventArgs e)
        {
            UserPreferences UP = new UserPreferences();
            try
            {

                if (Membership.MembershipHelper.CurrentUser != null)
                {
                    hdnUserId.Value = Membership.MembershipHelper.CurrentUser.ID.ToString();
                    ConnectionId.Value = Zamba.Membership.MembershipHelper.CurrentUser.ConnectionId.ToString();
                    hdnComputer.Value = Session["ComputerNameOrIP"].ToString();
                    //Page.Header.DataBind();
                    if (!Page.IsPostBack)
                    {
                        ZOptBusiness zopt = new ZOptBusiness();
                        string link = zopt.GetValue("WebHomeLink");
                        string target = zopt.GetValue("WebHomeTarget");
                        zopt = null;
                        hdnRefreshWF.Value = UP.getValue("WebRefreshWFTab", UPSections.WorkFlow, false);

                        if (string.IsNullOrEmpty(link))
                        {
                            hdnLink.Value = HttpContext.Current.Request.Url.ToString();
                        }
                        else
                        {
                            hdnLink.Value = link;
                        }

                        if (string.IsNullOrEmpty(target))
                        {
                            hdnTarget.Value = "_self";
                        }
                        else
                        {
                            hdnTarget.Value = target;
                        }
                        //Me.UC_WFExecution.TaskID = Task_ID

                        Session["ListOfTask"] = null;

                    }


                }

            }
            catch (Exception ex)
            {
                ZClass.raiseerror(ex);
            }
            finally { UP = null; }
        }


        protected void MasterLogout(object sender, System.EventArgs e)
        {
            Response.Redirect("~/Views/Security/Logout.aspx");
        }

        public static string GetAppRootUrl(bool endSlash)
        {
            string host = HttpContext.Current.Request.Url.GetLeftPart(UriPartial.Authority);
            string appRootUrl = HttpContext.Current.Request.ApplicationPath;
            if (!appRootUrl.EndsWith("/"))
            {
                //a virtual
                appRootUrl += "/";
            }
            if (!endSlash)
            {
                appRootUrl = appRootUrl.Substring(0, appRootUrl.Length - 1);
            }
            return host + appRootUrl;
        }

        public static string GetShowSessionRefreshLog()
        {
            SZOptBusiness zopt = new SZOptBusiness();
            string val = zopt.GetValue("ShowSessionRefreshLog");
            if (string.IsNullOrEmpty(val))
            {
                return "false";
            }
            else
            {
                return val.ToLower();
            }
        }
 
        //public static IHtmlString GetJqueryCoreScript()
        //{
        //    return Zamba.Web.Helpers.Tools.GetJqueryCoreScript(HttpContext.Current.Request);
        //}

        //public static string GetIsOldBrowser()
        //{
        //    return Zamba.Web.Helpers.Tools.GetIsOldBrowser(HttpContext.Current.Request).ToString().ToLower();
        //}
        public MasterPage()
        {
            Load += Page_Load;
            Init += Page_Init;
        }


    }
}
